src/services/formatting/formatting.ts

```diff
@@ -706,7 +706,6 @@ namespace ts.formatting {
                 if (isToken(child) && child.kind !== SyntaxKind.JsxText) {
                     // if child node is a token, it does not impact indentation, proceed it using parent indentation scope rules
                     const tokenInfo = formattingScanner.readTokenInfo(child);
-                    Debug.assert(tokenInfo.token.end === child.end, "Token end is child end");

```

Comment:
Finally tracked this down - the problem is we were creating a synthetic type reference node with text like `{ x: string }` when the type was anonymous, which was confusing the scanner because it expects a type reference node to point to an identifier or qualified name, so the formatting scanner and syntax tree disagreed about whether to expect a token. Fixed in extractMethod#485 and restored the assert